### PR TITLE
New version: Headlamp.Headlamp version 0.24.0

### PR DESCRIPTION
**Pull request by tianyeeT**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/157930)

### CODE REVIEW

File: manifests/h/Headlamp/Headlamp/0.20.1/Headlamp.Headlamp.installer.yaml [hunk 1]
```diff
@@ -15,7 +15,6 @@ Installers:
       SilentWithProgress: /S
     AppsAndFeaturesEntries:
       - DisplayName: Headlamp
-        DisplayVersion: 0.20.1
         Publisher: Headlamp
```
Comment (tianyeeT) [inline manifests/h/Headlamp/Headlamp/0.20.1/Headlamp.Headlamp.installer.yaml:18]:
```suggestion
        DisplayVersion: 0.20.1
        Publisher: Headlamp
```

File: manifests/h/Headlamp/Headlamp/0.21.0/Headlamp.Headlamp.installer.yaml [hunk 2]
```diff
@@ -15,7 +15,6 @@ Installers:
       SilentWithProgress: /S
     AppsAndFeaturesEntries:
       - DisplayName: Headlamp
-        DisplayVersion: 0.21.0
         Publisher: Headlamp
```
Comment (tianyeeT) [inline manifests/h/Headlamp/Headlamp/0.21.0/Headlamp.Headlamp.installer.yaml:18]:
```suggestion
        DisplayVersion: 0.21.0
        Publisher: Headlamp
```

File: manifests/h/Headlamp/Headlamp/0.22.0/Headlamp.Headlamp.installer.yaml [hunk 3]
```diff
@@ -15,7 +15,6 @@ Installers:
       SilentWithProgress: /S
     AppsAndFeaturesEntries:
       - DisplayName: Headlamp
-        DisplayVersion: 0.22.0
         Publisher: Headlamp
```
Comment (tianyeeT) [inline manifests/h/Headlamp/Headlamp/0.22.0/Headlamp.Headlamp.installer.yaml:18]:
```suggestion
        DisplayVersion: 0.22.0
        Publisher: Headlamp
```

File: manifests/h/Headlamp/Headlamp/0.23.0/Headlamp.Headlamp.installer.yaml [hunk 4]
```diff
@@ -15,7 +15,6 @@ Installers:
       SilentWithProgress: /S
     AppsAndFeaturesEntries:
       - DisplayName: Headlamp
-        DisplayVersion: 0.23.0
         Publisher: Headlamp
```
Comment (tianyeeT) [inline manifests/h/Headlamp/Headlamp/0.23.0/Headlamp.Headlamp.installer.yaml:18]:
```suggestion
        DisplayVersion: 0.23.0
        Publisher: Headlamp
```

File: manifests/h/Headlamp/Headlamp/0.23.1/Headlamp.Headlamp.installer.yaml [hunk 5]
```diff
@@ -15,7 +15,6 @@ Installers:
       SilentWithProgress: /S
     AppsAndFeaturesEntries:
       - DisplayName: Headlamp
-        DisplayVersion: 0.23.1
         Publisher: Headlamp
```
Comment (tianyeeT) [inline manifests/h/Headlamp/Headlamp/0.23.1/Headlamp.Headlamp.installer.yaml:18]:
```suggestion
        DisplayVersion: 0.23.1
        Publisher: Headlamp
```

File: manifests/h/Headlamp/Headlamp/0.23.2/Headlamp.Headlamp.installer.yaml [hunk 6]
```diff
@@ -15,7 +15,6 @@ Installers:
       SilentWithProgress: /S
     AppsAndFeaturesEntries:
       - DisplayName: Headlamp
-        DisplayVersion: 0.23.2
         Publisher: Headlamp
```
Comment (tianyeeT) [inline manifests/h/Headlamp/Headlamp/0.23.2/Headlamp.Headlamp.installer.yaml:18]:
```suggestion
        DisplayVersion: 0.23.2
        Publisher: Headlamp
```

**Comment by tianyeeT** [comment]
> Hi @tianyeeT,
> 
> This manifest removes Apps and Features entries that are present in previous PR versions. These entries should be added to this version, to maintain version matching, and prevent the 'upgrade always available' situation with this package.
> 
> (Automated response - build 885.)
> 
> Edit: This is because the previous version has:
> 
> ```yaml
>     AppsAndFeaturesEntries:
>       - DisplayName: Headlamp
>         DisplayVersion: 0.23.1
>         Publisher: Headlamp
> ```
> 
> `DisplayVersion` is an important one, potentially causing both issues with other manifests, and also the "upgrade always available' situation. But looking through [other PRs](https://github.com/search?q=repo%3Amicrosoft%2Fwinget-pkgs+Headlamp.Headlamp&type=pullrequests) for this package, all of them seem to have a `DisplayVersion` which matches their `PackageVersion` - if these always match, then `DisplayVersion` shouldn't be used.
> 
> For this PR to proceed, one of these should occur:
> 
> * The `DisplayVersion` removed from previous manifests.
> * The `DisplayVersion` added to this PR's manifest.
> 
> But only one and not both. The upper option is preferred, but is more work.

Thanks @stephengillie for your detailed explanation, and i will remove the `DisplayVersion` from previous manifests using separate PR.

**Comment by stephengillie** [comment]
Hi @tianyeeT,

This manifest removes Apps and Features entries that are present in previous PR versions. These entries should be added to this version, to maintain version matching, and prevent the 'upgrade always available' situation with this package.

(Automated response - build 886.)

**Review by stephengillie — APPROVED** [review]

**Pull request merged**